### ISSUE TITLE
connected  ' discoverServices '  is't get

### ISSUE THREAD

**Issue by LiangLuDev** [pauldemarco/flutter_blue#426]
Hi @pauldemarco 
```
List<BluetoothDevice> devices = await FlutterBlue.instance.connectedDevices;
List<BluetoothService> services= await devices[0].discoverServices();
print('$services');
```
Android  error  :
java.lang.IllegalArgumentException: Unsupported value: java.lang.Exception: no instance of BluetoothGatt, have you connected first?

iOS  error:
not print log and services

TKS

**Comment by miseyu** [comment]
me too.

**Comment by ride4sun** [comment]
@LiangLuDev yes the device has to be connected before you can call dicoverServices. The new version will throw an exception when you don't. Can this be closed?

**Comment by LiangLuDev** [comment]
@ride4sun current connected.
  iOS : Although the settings show that it is connected, it still needs to be connected in the app

**Comment by ride4sun** [comment]
Can describe that behavior a bit better? - Possibly with some code.

**Comment by woopluz** [comment]
same issuses.

**Comment by EricTURATSINZE** [comment]
Facing same issue
